Camera/CameraAlvium/camera_alvium.cc

```diff
@@ -129,37 +150,70 @@ void CameraAlvium::load_ini_params()
+    contrast_shape_ = pt.get<VmbInt64_t>("alvium.ContrastShape", contrast_shape_);
+    black_level = pt.get<double>("alvium.BlackLevel", black_level);
+    binning_horizontal_ = pt.get<VmbInt64_t>("alvium.BinningHorizontal", binning_horizontal_);
+    binning_horizontal_mode_ = pt.get<std::string>("alvium.BinningHorizontalMode", binning_horizontal_mode_);
+    binning_vertical_ = pt.get<VmbInt64_t>("alvium.BinningVertical", binning_vertical_);
+    binning_vertical_mode_ = pt.get<std::string>("alvium.BinningVerticalMode", binning_vertical_mode_);
+    adaptive_noise_suppression_factor_ =
+        pt.get<double>("alvium.AdaptiveNoiseSuppressionFactor", adaptive_noise_suppression_factor_);
 };
 
+#define VMB_IS_SET_OK(name, value) VMB_ERROR(camera_ptr_->GetFeatureByName(name, fp)) || VMB_ERROR(fp->SetValue(value))
+
 void CameraAlvium::bind_params()
 {
-    VmbCPP::FeaturePtr fp; // Generic feature pointer
-
-    /* TODO : Add more params, check with mickael
-        \A -> \B => VMB_ERROR(camera_ptr_->GetFeatureByName(\A, fp)) || VMB_ERROR(fp->SetValue(\B))
-
-        "DeviceLinkThroughputLimit" -> 450'000'000
-        "AcquisitionFrameRateEnable" -> "true"
-        "AcquisitionFrameRate" -> 10
-        "ExposureAuto" -> "Off"
-        "ExposureTime" -> EXP (I dont know what it is need to check API Manuel)
-        "GainAuto" -> "Off"
-        "SensorBitDepth" -> "Bpp12"
-        "OffsetX" -> X0
-        "OffsetY" -> Y0
-    */
-
-    if (VMB_ERROR(camera_ptr_->GetFeatureByName("PixelFormat", fp)) || VMB_ERROR(fp->SetValue("Mono8")) ||
-        VMB_ERROR(camera_ptr_->GetFeatureByName("Width", fp)) || VMB_ERROR(fp->SetValue(width_)) ||
-        VMB_ERROR(camera_ptr_->GetFeatureByName("Height", fp)) || VMB_ERROR(fp->SetValue(height_)))
+
+    VmbCPP::FeaturePtr fp; // Generic feature pointer use inside VMB_IS_SET_OK macro
+
+    if (VMB_IS_SET_OK("Width", width_) || VMB_IS_SET_OK("Height", height_) || VMB_IS_SET_OK("PixelFormat", "Mono8") ||
+        VMB_IS_SET_OK("ReverseX", reverse_x_) || VMB_IS_SET_OK("ReverseY", reverse_y_) ||
+        VMB_IS_SET_OK("Gamma", gamma_) || VMB_IS_SET_OK("Gain", gain_) ||
+        VMB_IS_SET_OK("LensShadingValue", lens_shading_value_) ||
+        VMB_IS_SET_OK("IntensityAutoPrecedence", intensity_auto_precedence_.c_str()) ||
+        VMB_IS_SET_OK("ExposureActiveMode", exposure_active_mode_.c_str()) ||
+        VMB_IS_SET_OK("ExposureAuto", exposure_auto_.c_str()) || VMB_IS_SET_OK("ExposureTime", exposure_time_) ||
+        VMB_IS_SET_OK("CorrectionMode", correction_mode_.c_str()) ||
+        VMB_IS_SET_OK("CorrectionSelector", correction_selector_.c_str()) ||
+        VMB_IS_SET_OK("ContrastBrightLimit", contrast_bright_limit_) ||
+        VMB_IS_SET_OK("ContrastDarkLimit", contrast_dark_limit_) || VMB_IS_SET_OK("ContrastEnable", contrast_enable_) ||
+        VMB_IS_SET_OK("ContrastShape", contrast_shape_) || VMB_IS_SET_OK("BlackLevel", black_level) ||
```

Comment:
Maybe add a _ to the end of the 'black_level' variable ? (ye im like that sorry)